chemprop/nn/loss.py

```diff
@@ -66,13 +71,13 @@ def __call__(
         Tensor
             a scalar containing the fully reduced loss
         """
-        L = self.forward(preds, targets, mask, w_s, w_t, lt_mask, gt_mask)
-        L = L * w_s.view(-1, 1) * w_t.view(1, -1) * mask
+        L = self._calc_unreduced_loss(preds, targets, mask, w_s, lt_mask, gt_mask)
+        L = L * w_s.view(-1, 1) * self.w_t.view(1, -1) * mask
```

Comment:
what if `self.w_t = None`? This will raise an error

Reply:
`w_t` should never be `None` if the `LossFunction` is being used as a loss function. But `Metric` inherits the `__init__` from `LossFunction` even though it does not need `w_t`, so there has to be a default. Another solution would be do decouple `Metric` from `LossFunction` by defining separate `__init__` functions for them, but that would require some code duplication in metrics like `BCEMetric`. 

If the user defines their own criterion to give to predictor, then I think it is reasonable for them to read the doc string "`None` option is for when the function is used as a metric.", especially after getting the error:
```
---> [75] L = L * w_s.view(-1, 1) * self.w_t.view(1, -1) * mask
AttributeError: 'NoneType' object has no attribute 'view'
```
At that point they know it is either `w_s` or `w_t` that is the problem.

Reply:
We could also just mention in the `Metric` docstring that the variable is unused rather than specifying that the variable be `None`. The current type hint implies that `None` is a valid choice which conflicts with the docstring and actual usage.

Reply:
re: `AttributeError`

True, the error is very clear what the problem is, but this seems avoidable, no? Why should we wait for a user to try and calculate a loss value to tell them that've parameterized it incorrectly when we know _by construction_ what a valid choice is?

Reply:
> We could also just mention in the `Metric` docstring that the variable is unused rather than specifying that the variable be `None`. The current type hint implies that `None` is a valid choice which conflicts with the docstring and actual usage.

This would work, but does require the user to pass something that can be turned into a tensor when creating a Metric, even though that something will be entirely unused. But this does have the benefit that it fails earlier, i.e. when the Metric is created if the user doesn't give something for `w_t` or gives `None`.